pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -402,6 +402,40 @@ func (gc *GarbageCollector) runAttemptToOrphanWorker() {
 	}
 }
 
+func (gc *GarbageCollector) isSafeToOrphan(n *node) (safe bool, reason string) {
+	latest, err := gc.getObject(n.identity)
+	switch {
+	case errors.IsNotFound(err):
+		glog.V(5).Infof("item %v not found, go ahead orphaning its dependents anyway", n.identity)
+		return true, ""
+	case err != nil:
+		err2 := fmt.Errorf("cannot get object %v: %v", n.identity, err)
+		utilruntime.HandleError(err2)
+		return false, err2.Error()
+	}
+	generation := latest.GetGeneration()
+	// if generation is 0, we assume it's because no controller
+	// sets the generation for this object (like "pod"), so no need to
+	// synchronize with the controller and it's safe to proceed with the
+	// orphaning procedures.
+	if generation == 0 {
+		return true, ""
+	}
+	observedGeneration := latest.GetObservedGeneration()
```

Comment:
Controllers are not wiring refs for objects pending deletion - remember though that they are reading from a cache. Assuming that DELETE bumps generation, we need the GC to wait for the deletion to be noticed by the owning controller before proceeding here.